server/src/main/java/csearch/controller/CsController.java

```diff
@@ -83,15 +93,19 @@ public CsProject update(@PathVariable(value = "id") Integer projectId, @RequestB
 
     CsProject project = this.csRepo.findById(projectId).get();
 
-    if (title != null && !title.isEmpty()) project.setTitle(title);
-    if (description != null) project.setDescription(description);
-    if (process != null) project.setProcess(process);
+    if (title != null && !title.isEmpty())
```

Comment:
You already do these validation checks in CsProject model. So you do not need to do these checks. See below:
project.setTitle(title);
project.setDescription(description);
...

Reply:
true for all but difficulty